lib/importers/licence_transaction/licence_importer.rb

```diff
@@ -80,11 +85,35 @@ def licences
         )["results"]
       end
 
-      def filtered_licences
-        licence_identifiers = File.new(common_licences_path).readlines(chomp: true)
+      def licences_tagging
+        @licences_tagging ||= grouped_licences.map do |licence_url, rows|
+          {
+            "base_path" => URI.parse(licence_url).path,
+            "locations" => locations(rows).uniq,
+            "industries" => industries(rows).uniq,
+          }
+        end
+      end
+
+      def licence_tagging(base_path)
+        licences_tagging.find { |l| l["base_path"] == base_path }
+      end
 
-        licences.select do |licence|
-          licence_identifiers.include?(licence["details"]["licence_identifier"])
+      def locations(grouped_licence)
+        grouped_licence.flat_map do |licence|
+          licence["Locations"].split(", ").map(&:parameterize)
+        end
+      end
+
+      def industries(grouped_licence)
+        grouped_licence.flat_map do |licence|
+          licence[2..].compact.map(&:parameterize)
+        end
+      end
+
+      def grouped_licences
+        CSV.foreach(licence_tagging_path, headers: true).group_by do |licence|
+          licence["Link"]
```

Comment:
So are there duplicate rows, or is this just to mitigate? Does that mean we're tagging to all of the industries in both rows, rather than overwriting the first row?

Reply:
There are ‘duplicate’ rows in the spreadsheet due to different licences actually pointing to the same licence page (intentionally). As I’ve not tried to modify the spreadsheet too much manually, the code takes this into consideration. 

The approach I took is just to add all the unique industries together for each link.

Reply:
Out of curiosity, do you have an example? I'd love to see it in action.

I assume the content designers are aware of the duplication? It's something they may want to look into.

Reply:
I've let Jessie know. I did spot checked the duplicates and they looked reasonable as either a licence page covers two licences (like https://www.gov.uk/licences-use-music-online-mobile) or there are separate rows per location (like https://www.gov.uk/greenhouse-gas-emissions-permit). The actual licence list came from a BEIS audit, rather than something we generated so I'd expect (hope) they'd be correct, we just added the proposed industries for each licence.

I've just pulled the links that have more than one row below, you can search for the links in the [spreadsheet](https://docs.google.com/spreadsheets/d/1ZfGKnVKmfB5yGBWMx3PvwKq8KEj6Lt1XzcWXCJWBMf0/edit#gid=1760190301) under the 'Matching industries to BEIS's licence audit' tab.

```
=> ["https://www.gov.uk/approval-and-accreditation-of-vocational-training-courses-scotland",
 "https://www.gov.uk/gas-electricity-supply-licence",
 "https://www.gov.uk/scheduled-monument-consent-england-scotland-wales",
 "https://www.gov.uk/greenhouse-gas-emissions-permit",
 "https://www.gov.uk/fireworks-manufacture-registration",
 "https://www.gov.uk/licences-use-music-online-mobile",
 "https://www.gov.uk/registration-of-independent-health-care-service-england-wales",
 "https://www.gov.uk/second-hand-dealers-registration-scotland"]
```